Migrated to hilt

mario219 opened the pull request:
Migrated DI to Hilt

Commit:
Migrated to hilt

Commit:
- Injecting Dispatchers

eis6aer commented on data/src/main/java/com/example/data/datasource/PostsRemoteDataSourceImpl.kt:
```diff
@@ -19,13 +21,14 @@ internal class PostsRemoteDataSourceImpl @Inject constructor(
     private val commentsMapper: BaseRemoteMapper<List<CommentsRemote>, List<Comments>>,
-    private val userMapper: BaseRemoteMapper<UserRemote, User>
+    private val userMapper: BaseRemoteMapper<UserRemote, User>,
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher
```
The only comment I would make is to change the name of the parameter to not tie it to an ioDispatcher

eis6aer commented on data/src/main/java/com/example/data/di/DataSourceModule.kt:
```diff
@@ -6,20 +6,19 @@ import com.example.domain.datasource.PostsLocalDataSource
 
     @Binds
-    @Singleton
+    @Reusable
```
Why would you make this reusable instead of SIngleton, this change is good but it is important to understand the difference

eis6aer commented on data/src/main/java/com/example/data/di/DispatchersModule.kt:
```diff
@@ -0,0 +1,30 @@
+    fun providesMainDispatcher(): CoroutineDispatcher = Dispatchers.Main
+}
+
+@Retention(AnnotationRetention.BINARY)
```
Probably the annotations would be much better in another class

Commit:
- Injecting Dispatchers

Commit:
[Making the app runs again]

Commit:
[Updated this repo to make it run again]

- Converted domain module from Java library to Android Library
- Fixed dependencies gradle file in buildsystem
- Fixed network calls error handler
- Some other small improves through the clean arch impl
- Added UTs for:
 _Data layer_
 * ApiResponseHandler.kt
 -Domain layer-
 * GetPostOwnerUseCase.kt
 * GetCommentsUseCase.kt
 * RetrievePostsFromRemoteUseCase.kt
 _Presentation layer_
 * DetailViewModel.kt

mariodev219 committed:
Update README.md
